src/server/bitops_family_test.cc

```diff
@@ -126,4 +284,166 @@ TEST_F(BitOpsFamilyTest, BitCountByteBitSubRange) {
   EXPECT_EQ(0, CheckedInt({"bitcount", "foo", "-1", "-2", "bit"}));  // illegal range
 }
 
+// for the bitop tests we need to test with multiple keys as the issue
+// is that we need to make sure that accessing multiple shards creates
+// the correct result
+// Since this is bit operations, we are using the bytes data type
+// that makes the verification more ergonomics.
+const std::pair<std::string_view, Bytes> KEY_VALUES_BIT_OP[] = {
+    {"first_key", 0xFFAACC01_b},
+    {"key_second", {0x1, 0xBB}},
+    {"_this_is_the_third_key", {0x01, 0x05, 0x15, 0x20, 0xAA, 0xCC}},
+    {"the_last_key_we_have", 0xAACC_b}};
+
+TEST_F(BitOpsFamilyTest, BitOpsAnd) {
+  auto resp = Run({"bitop", "foo", "bar", "abc"});  // should failed this is illegal operation
+  ASSERT_THAT(resp, ErrArg("syntax error"));
+  // run with none existing keys, should return 0
+  EXPECT_EQ(0, CheckedInt({"bitop", "and", "dest_key", "1", "2", "3"}));
+
+  resp = Run({"set", KEY_VALUES_BIT_OP[0].first, KEY_VALUES_BIT_OP[0].second});
+  EXPECT_EQ(resp, "OK");
+  EXPECT_EQ(KEY_VALUES_BIT_OP[0].second.size(),
+            CheckedInt({"bitop", "and", "foo_out", KEY_VALUES_BIT_OP[0].first}));
+
+  auto res = Bytes::From(Run({"get", "foo_out"}));
+  EXPECT_EQ(res, KEY_VALUES_BIT_OP[0].second);
+  // this will 0 all values other than one bit it would end with result with length ==
+  //     FOO_KEY_VALUE && value == BAR_KEY_VALUE
+
+  resp = Run({"set", KEY_VALUES_BIT_OP[1].first, KEY_VALUES_BIT_OP[1].second});
+  EXPECT_EQ(resp, "OK");
+  const auto EXPECTED_LEN =
+      std::max(KEY_VALUES_BIT_OP[0].second.size(), KEY_VALUES_BIT_OP[1].second.size());
+  EXPECT_EQ(EXPECTED_LEN, CheckedInt({"bitop", "and", "foo-out", KEY_VALUES_BIT_OP[0].first,
+                                      KEY_VALUES_BIT_OP[1].first}));
+  const auto EXPECTED_RESULT = Bytes((0xffaacc01 & 0x1BB));  // first and second values
```

Comment:
pls add empty lines between each test, right now it's a bit hard to separate those visually

Reply:
OK

Reply:
This will take a little longer to make sure that this is more readable - so I will push again later than today

Reply:
All the tests were re-writing with all the shared code moved out, and the actions are now marked with what each step is doing (basically its checking whether val1 op .. valn output from the programming language match the actual value we've assigned to.